### PR TITLE
Improve daily by discarding items already processed

### PR DESCRIPTION
**Pull request by fernandog**
### Story
As parsing release names is one of the most resource intense actions, i'd like to reduce the processing of results drastically through not parsing items that are already parsed and located in the providers cache tables.

### Goals:
- Run dailysearch faster. Today it takes 1-2 minutes per provider
- Avoid reprocess (parse) same item over and over
- Optional per provider. If provider doesn't support `pubdate` we don't add the check
- Small changes in code (clean solution). Change only what is necessary to increase time-to-market

### Requirements
- Reduce CPU load and Memory usage during daily provider searches
- Reduce daily search duration, freeing up resources as soon as possible.
- .. 

### Solution

- Store newest pubdate from provider in `lastPubdate` table using the same solution from provider "lastSearch"

image

### Test Results
Results:
+- 83% of dailysearch execution time reduction (I have trusted enabled) - 10s per provider
+-50% for @duramato 

I'm storing  `last_pubdate` from provider in lastPubdate table (similar to lastSearch table logic)

@pymedusa/developers @pymedusa/owners 
Need someone to find **logic flaw** in this PR (review logic). 
Review code can be later (duplicate stuff and etc)
So far we haven't missed any item in daily

- [x] Rarbg
- [X] Kat
- [X] MoreThanTv
- [X] Alpharatio
- [X] Lime
- [X] Transmithenet
- [X] Danishbits
- [X] Hounddawgs
- [X] Scenelelite
- [X] Torrentz
- [X] Extratorrent
- [X] HD4free
- [X] HDTorrents
- [X] Speed.cd
- [X] newznab (including torznab)
- [X] TorrentDay
- [X] TorrentBytes

Needs rewrite first:
- [ ] anizb - item.pubDate.get_text(strip=True)
- [ ] binsearch - pubdate = item.get('published')
- [ ] xthor - need to use API
- [ ] btn

Pending:
- [ ] abnormal
- [ ] bitcannon
- [ ] bitsnoop
- [ ] bitsoup
- [ ] bluetigers
- [ ] btdigg
- [ ] cpasbien
- [ ] elitetorrent
- [ ] freshontv
- [ ] gftracker
- [ ] hdbits
- [ ] hdspace
- [ ] ilovetorrents
- [ ] iptorrents
- [ ] newpct
- [ ] norbits
- [ ] nyaatorrents
- [ ] omgwtfnzbs
- [ ] pretome
- [ ] rsstorrent
- [ ] scc
- [ ] scenetime
- [ ] shazbat
- [ ] t411
- [ ] thepiratebay
- [ ] tntvillage
- [ ] tokyotoshokan
- [ ] torrentleech
- [ ] torrentproject
- [ ] tvchaosuk
- [ ] womble

### CODE REVIEW

File: sickrage/helper/common.py [hunk 1]
```diff
@@ -328,3 +328,21 @@ def enabled_providers(search_type):
...
+    :param old_string: String that will be cleaned
+    :param unwanted_strings: List of unwanted strings
+
+    :returns: the string without the unwanted strings
+    """
+    if not old_string:
+        return
+
+    for item in unwanted_strings:
+        new_string = old_string.replace(item, '') if isinstance(item, str) else new_string
```
Comment (p0psicles) [inline sickrage/helper/common.py]: your not replacing them incrementally.
Just do: old_string = old_string.replace(item, '')

Also don't need to type check there. Your passing the list with strings your self. You can expect it to always have the correct value right?

File: sickbeard/providers/alpharatio.py [hunk 2]
```diff
@@ -175,6 +185,13 @@ def process_column_header(td):
             items.sort(key=lambda d: try_int(d.get('seeders', 0)), reverse=True)
             results += items
 
+            # Set last pubdate for provider
+            if results and mode == 'RSS':
+                last_pubdate =  max([result['pubdate'] for result in results])
```
Comment (p0psicles) [inline sickbeard/providers/alpharatio.py]: space before max

File: sickbeard/providers/danishbits.py [hunk 3]
```diff
@@ -146,6 +152,13 @@ def process_column_header(td):
 
                             seeders = try_int(cells[labels.index('Seeders')].get_text(strip=True))
                             leechers = try_int(cells[labels.index('Leechers')].get_text(strip=True))
+                            pubdate = cells[labels.index('Tilføjet')].find('span')['title']
+                            pubdate = parser.parse(pubdate, fuzzy=True)
+
+                            # Here we discard item if is not a new item
+                            if pubdate and last_pubdate and mode == "RSS" and pubdate < last_pubdate:
+                                #logger.log("Discarded {0} because it was already processed. Pubdate: {1}".format(title, pubdate))
```
Comment (p0psicles) [inline sickbeard/providers/danishbits.py]: Space between # and logger please. Fix for all prov.

File: sickbeard/providers/alpharatio.py [hunk 4]
```diff
@@ -158,7 +158,7 @@ def process_column_header(td):
     
                             # Here we discard item if is not a new item
                             if pubdate and last_pubdate and mode == "RSS" and pubdate < last_pubdate:
-                                #logger.log("Discarded {0} because it was already processed. Pubdate: {1}".format(title, pubdate))
+                                # logger.log("Discarded {0} because it was already processed. Pubdate: {1}".format(title, pubdate))
```
Comment (fernandog) [inline sickbeard/providers/alpharatio.py]: @p0psicles

**Comment by fernandog** [comment]
@p0psicles are you still testing this?

File: sickbeard/tvcache.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def __init__(self, providerName):
             if not self.hasTable(providerName):
                 logger.log(u"Creating cache table for provider {}".format(providerName), logger.DEBUG)
                 self.action(
-                    "CREATE TABLE [" + providerName + "] (name TEXT, season NUMERIC, episodes TEXT, indexerid NUMERIC, url TEXT, time NUMERIC, quality TEXT, release_group TEXT)")
+                    "CREATE TABLE [" + providerName + "] (name TEXT, season NUMERIC, episodes TEXT, indexerid NUMERIC, url TEXT, time NUMERIC, quality NUMERIC, release_group TEXT)")
```
Comment (fernandog) [inline sickbeard/tvcache.py]: @p0psicles testing in pubdate2 but I wont delete my cache.db so this wont change for me

File: sickbeard/tvcache.py [hunk 6]
```diff
@@ -308,24 +342,16 @@ def shouldClearCache(self):
 
         return False
 
-    def _addCacheEntry(self, name, url, seeders, leechers, size, pubdate, hash, parse_result=None, indexer_id=0):
```
Comment (fernandog) [inline sickbeard/tvcache.py]: `parse_result` and `indexer_id=0` weren't used anywhere

File: sickbeard/tvcache.py [hunk 7]
```diff
@@ -308,24 +342,16 @@ def shouldClearCache(self):
...
-            if indexer_id:
-                showObj = Show.find(sickbeard.showList, indexer_id)
-
-            try:
-                parse_result = NameParser(showObj=showObj).parse(name)
-            except (InvalidNameException, InvalidShowException) as error:
-                logger.log(u"{}".format(error), logger.DEBUG)
-                return None
+        try:
+            parse_result = NameParser().parse(name)
```
Comment (fernandog) [inline sickbeard/tvcache.py]: this fixed an issue from @duramato about not parsing correctly a show with scene exception

**Comment by fernandog** [comment]
@p0psicles feature branch is clean now.

**Pull request closed without merging**